server/src/main/java/com/google/speech/tools/voxetta/servlets/BlobstoreLinkServlet.java

```diff
@@ -1,48 +1,52 @@
 // Copyright 2020 Google LLC
- 
+//
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
- 
+// 
 //     https://www.apache.org/licenses/LICENSE-2.0
- 
+//
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
  
-package com.google.speech.tools.voxetta;
- 
-import com.google.appengine.api.blobstore.BlobstoreService;
-import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+package com.google.speech.tools.voxetta.servlets;
+
+import com.google.appengine.api.blobstore.BlobstoreFailureException;
+import com.google.speech.tools.voxetta.services.DatastoreUtteranceService;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
  
 /** 
-* Servlet that returns a Blobstore URL. 
-*/
-@WebServlet("/blobstore-link")
+ * Servlet that returns a Blobstore URL. 
+ */
+@WebServlet("/blobstore-utterance-upload-link")
 public class BlobstoreLinkServlet extends HttpServlet {
- 
-  private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+
+  private DatastoreUtteranceService service = new DatastoreUtteranceService(); 
  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Redirect to Datastore processing after Blobstore processing
-    String uploadUrl = blobstoreService.createUploadUrl("/upload-utterance");
- 
-    response.setContentType("text/html");
-    response.getWriter().println(uploadUrl);
+    response.setContentType("application/json");
+
+    // Create and return a Blobstore Upload URL  
+    try {
+      String uploadUrl = service.getFormUrl();
+      response.getWriter().println("{ \"success\": true, \"url\": \"" + uploadUrl + "\" }"); 
```

Comment:
Can you use the GSon library to serialize this? Doesn't it figure out the fields and create an object out of it?

Simple class like this would be a good general purpose utility, reusable throughout many servlets.

```java
public class StatusResponse {
  private boolean success;

  public StatusResponse(boolean success) {
    this.success = success;
  }

  // not sure on type that GSOn creates?
  public String toJson() {
    return GSON.toJson(this);
  }
}
```

This helps avoid the messy string concatenation, adds some nice type checking to your responses. Looks like you have several flavors of responses here, you could easily create subclasses such as UrlResponse, ErrorResponse which have different fields.

example subclass

```java
public class ErrorResponse extends StatusResponse {
  private String error;
  public ErrorResponse(boolean success, String error) {
    super(success);
    this.error = error;
   }

   // also inherits toJson from StatusResponse
}
```

Reply:
I will add this as a TODO and work on it immediately after merging this pr!